### PR TITLE
[Feat] tuple type

### PR DESCRIPTION
**Pull request by jay-tux**
#### Description of Change
Added a (naive) implementation of C++'s `std::tuple`.  
However, some of the things required (`if constexpr`, `auto` return type) require C++17.

#### Checklist
- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: requires C++17 (hence the version bump in the master CMakeLists.txt); however, this didn't break anything (`cmake .. && cmake --build .` in a `/build/` succeeded)

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1809"></a>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ project(Algorithms_in_C++
 # set(CMAKE_CXX_CPPLINT "~/anaconda3/bin/cpplint --filter=-legal/copyright --std=c++11")
 # find_program(CLANG_FORMAT "clang-format")
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (Panquesito7) [inline CMakeLists.txt:11]: Wouldn't we have to update all the other codes to match the C++-17 syntax?

Comment (jay-tux) [reply CMakeLists.txt:11]: I changed the general CMake file; I'll check any other now and update them all

Comment (jay-tux) [reply CMakeLists.txt:11]: Seems there are no others which define their own C++ version.  
Compilation threw a warning from another algorithm, yet it seems that this should've existed before (no return in non-void)

Comment (Panquesito7) [reply CMakeLists.txt:11]: So what should we do, then? I'm confused. 🤔

Comment (Panquesito7) [reply CMakeLists.txt:11]: > I think the issue is in the CI: could it be that it is set to use C++11?

Hmm, I think it's due to that, not sure though. Maybe we should try to make it work in C++11.

**Pull request closed without merging**